tests/utils/net_test.py

```diff
@@ -60,6 +60,12 @@ def test_gcs_object_url(self, seed):
         assert net._build_gcs_object_url(bucket_name, bucket_path) == \
                "https://storage.googleapis.com/storage/v1/b/unittest-bucket.test.me/o/path%2Fto%2Fobject?alt=media"
 
+    def test_add_url_param(self):
+        url = "https://artifacts.elastic.co/downloads/elasticsearch/elasticsearch-7.2.0.tar.gz?flag1=true"
```

Comment:
Should we add a second test to check if a flag value contains something that needs special quoting e.g. a whitespace? Reason I am saying this is because I noticed that [urlencode](https://docs.python.org/3.8/library/urllib.parse.html#urllib.parse.urlencode) by default uses `quote_plus` which e.g. will encode a param like `flag=test me` to `flat=test+me`. The plus is OK for queries according to http://en.wikipedia.org/wiki/URL_encoding and http://www.w3.org/TR/html401/interact/forms.html#h-17.13.4.1 but perhaps having this test helps us figure out some edge case in the future?